api/src/identity-access-management/domain/models/ClaimsMapper.js

```diff
@@ -0,0 +1,50 @@
+import flatten from 'lodash/flatten.js';
+import pick from 'lodash/pick.js';
+
+import { DEFAULT_CLAIMS_MAPPING } from '../constants/oidc-identity-providers.js';
+
+const DEFAULT_OPTIONS = { mapping: DEFAULT_CLAIMS_MAPPING, additionalClaims: [] };
+
+export class ClaimsMapper {
+  constructor(options) {
+    const { mapping = {}, additionalClaims = [] } = { ...DEFAULT_OPTIONS, ...(options || {}) };
+
+    this.additionalClaims = additionalClaims;
+
+    this.mapping = {
+      ...mapping,
+      ...additionalClaims.reduce((result, claim) => ({ ...result, [claim]: [claim] }), {}),
+    };
+  }
+
+  map(userInfo = {}) {
+    return Object.entries(this.mapping).reduce((result, [userKey, claimKeys]) => {
+      for (const claimKey of claimKeys) {
+        if (userInfo?.[claimKey]) {
+          return { ...result, [userKey]: userInfo[claimKey] };
+        }
+      }
+      return result;
+    }, {});
+  }
+
+  mapAdditionalClaims(userInfo = {}) {
+    return pick(userInfo, this.additionalClaims);
+  }
+
+  isMissingClaims(userInfo = {}) {
```

Comment:
Petite suggestion pour rendre la lecture du code plus naturelle : 
```suggestion
  hasMissingClaims(userInfo = {}) {
```